sdk/core/azure-core-amqp/src/main/java/com/azure/core/amqp/implementation/ReactorReceiver.java

```diff
@@ -66,14 +64,15 @@ protected ReactorReceiver(String entityPath, Receiver receiver, ReceiveLinkHandl
                     }
                 }
             })
-            .subscribeWith(EmitterProcessor.create());
+            .publish()
+            .autoConnect();
         this.endpointStates = this.handler.getEndpointStates()
             .map(state -> {
                 logger.verbose("connectionId[{}], path[{}], linkName[{}]: State {}", handler.getConnectionId(),
                     entityPath, getLinkName(), state);
                 return AmqpEndpointStateUtil.getConnectionState(state);
             })
-            .subscribeWith(ReplayProcessor.cacheLastOrDefault(AmqpEndpointState.UNINITIALIZED));
+            .cache(1);
```

Comment:
cache(1) works a little differently than cacheLast which was previously used. `cache(n)` emits the last n items emitted by the source when a subscription is requested even if the source hasn't emitted a complete signal. This doesn't guarantee that the emitted signal is the last signal from the source.

Reply:
I thought cacheLast was replay the last n items. I must have misunderstood it... phew. glad that was fixed.